pkg/controllers/binding/binding_controller.go

```diff
@@ -275,8 +276,11 @@ func (c *ResourceBindingController) updateFullyAppliedCondition(binding *workv1a
 		Message: FullyAppliedSuccessMessage,
 	}
 
-	meta.SetStatusCondition(&binding.Status.Conditions, newBindingFullyAppliedCondition)
-	err := c.Client.Status().Update(context.TODO(), binding)
-
-	return err
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() (err error) {
+		if err = c.Get(context.TODO(), client.ObjectKey{Namespace: binding.Namespace, Name: binding.Name}, binding); err != nil {
+			return err
+		}
+		meta.SetStatusCondition(&binding.Status.Conditions, newBindingFullyAppliedCondition)
```

Comment:
does it need to call function `SetStatusCondition` every time Retry? Is this specified? 😄

Reply:
we should make sure Get operation is successful.